core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -431,6 +446,19 @@ public void start() {
         logger().info("Docker Compose has finished running");
 
         AuditLogger.doComposeLog(this.getCommandParts(), this.getEnv());
+
+        final Integer exitCode = this.dockerClient.inspectContainerCmd(containerId)
+                .exec()
+                .getState()
+                .getExitCode();
+
+        if (exitCode == null || exitCode != 0) {
+            throw new ContainerLaunchException(
```

Comment:
As with `LocalDockerCompose`, we now throw an exception if a failure occurred. This allows us to detect a failure of `docker-compose down`.